gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import torch
+
+from ..utils import cached
+from .added_diag_lazy_tensor import AddedDiagLazyTensor
+from .diag_lazy_tensor import DiagLazyTensor
+from .kronecker_product_lazy_tensor import KroneckerProductLazyTensor
+from .non_lazy_tensor import lazify
+
+# TODO: check the speed of the linear algebra - svd vs eig etc.
+
+
+class KroneckerProductAddedDiagLazyTensor(AddedDiagLazyTensor):
+    def __init__(self, *lazy_tensors):
+        super(KroneckerProductAddedDiagLazyTensor, self).__init__(*lazy_tensors)
+        if len(lazy_tensors) > 2:
+            raise RuntimeError("An AddedDiagLazyTensor can only have two components")
```

Comment:
this seems odd - why not explicitly require the tensors so there is no ambiguity? 

```
def __init__(self, kron: KroneckerLazyTensor, diag DiagLazyTensor):
   ...
```

Reply:
one potential issue is that you want this tensor to be called whenever you add `KroneckerLazyTensor` and `DiagLazyTensor` so I'd have to go through the codebase checking this?

Reply:
I think we can mostly handle this in `__add__` of both `KroneckerLazyTensor`and `DiagLazyTensor`?